### ISSUE TITLE
Perhaps mentioning the datatype to be float for the tensors in documentation could be beneficial

### ISSUE THREAD

**Issue by tvmsandy33** [tatp22/multidim-positional-encoding#39]
while doing PositionalEncoding2D(1), I encountered the behavior that output tensor was equal to input tensor. After some fiddling, I figured that changing the dtype of the input tensor to float solved the issue. Perhaps this info could be added to the documentation.

**Comment by tatp22** [comment]
Hey @tvmsandy33,

Could you give me some code snippets to reproduce this behavior? I can add a quick "gotchas" section to the readme.

**Comment by tvmsandy33** [comment]
This is particularly the case with single channel inputs (i.e output tensor is equal to input tensor). Though results for multi-channel int dtypes is also not as expected.

```python
x = torch.randint(0,9,size=(1,3,3,1),dtype=int)

pos = Summer(PositionalEncoding2D(1))

print(x)
pos(x)
```

![git1](https://github.com/user-attachments/assets/59047048-f03c-4145-89c8-cf4a08a5182c)

Whereas converting the dtype to be float solves this issue.

```python
x = torch.randint(0,9,size=(1,3,3,1),dtype=float)

pos = Summer(PositionalEncoding2D(1))

print(x)
pos(x)
```

![git2](https://github.com/user-attachments/assets/703624e0-432d-4ec2-863d-8922a6c6343d)

**Comment by tatp22** [comment]
Ah, I think i see what the issue might be.

When we run the forward pass with the positional encodings, we create the embedding tensor, which takes the same dytpe as the input tensor:

```python
emb = torch.zeros((x, self.channels), device=tensor.device, dtype=tensor.dtype)
emb[:, : self.channels] = emb_x
```

However the values that the positional encoding tensor can take are strictly between [-1, 1]. When we try to cast a float dtype in this range to an int, it always goes to 0:

```python
>>> x = torch.tensor([0.7, 0.6, -0.6], dtype=int)
>>> x
tensor([0, 0, 0])
```

So that's why there's no positional encodings added to the tensor at all, since all values on the 0th channel dimension start at 0 (the formulas states that `PE(x,y,0) = sin(x)` and since the channel dimension is always 0 that means that this will always be (-1, 1) since the only time `sin(x) = 1` or `sin(x) = -1` is on any multiple of pi/2, which is never an int :shrug:), thus the 0 tensor will always be added on the 0th channel dimension.

Anyways, I think that the fix would be to make the embedding tensor to always be a float, or at least when it's an int to cast it into a float. I can create a change for that and send it out.

Thanks for finding the bug!

**Comment by tatp22** [comment]
Fixed in #40
